src/utils/extract.js

```diff
@@ -63,5 +63,22 @@ export const extract = {
         });
   
       return clusteredAverages;
+    },
+    rgbToHexadecimal: (r,g,b) => {
+        let redHex = r.toString(16);
+        let greenHex = g.toString(16);
+        let blueHex = b.toString(16);
+
+        if (redHex.length === 1) {
+            redHex = '0' + redHex;
+        }
+        if (greenHex.length === 1) {
+            greenHex = '0' + greenHex;
+        }
+        if (blueHex.length === 1) {
+            blueHex = '0' + blueHex;
+        }
+
+        return '#' + redHex + greenHex + blueHex;
```

Comment:
Could we use string interpolation here? return \`#${redHex}${greenHex}${blueHex}\`;

Reply:
concatenation and template literals are interchangeable here. this feels more readable with concatenation, which might be the only case I would prefer concatenation over template literals. as this is mostly variables and barely any strings, and doesn't save any space. could go either way honestly.

Reply:
I'd buy that. Readable is always better.